connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala

```diff
@@ -205,14 +220,26 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
    * Called only by SparkConnectSessionManager.
    */
   private[connect] def close(): Unit = {
+    // It is called only by SparkConnectSessionManager.closeSession, only after it's removed from
+    // the sessionStore there guarantees that it is called only once.
+    if (closedTime.isDefined) {
+      throw new IllegalStateException(s"Session $key is already closed.")
+    }
+
     logInfo(s"Closing session with userId: $userId and sessionId: $sessionId")
 
-    // After isClosing=true, SessionHolder.addExecuteHolder() will not allow new executions for
-    // this session. Because both SessionHolder.addExecuteHolder() and
+    // After closedTime is defined, SessionHolder.addExecuteHolder() will not allow new executions
+    // to be added for this session. Because both SessionHolder.addExecuteHolder() and
     // SparkConnectExecutionManager.removeAllExecutionsForSession() are executed under
     // executionsLock, this guarantees that removeAllExecutionsForSession triggered below will
     // remove all executions and no new executions will be added in the meanwhile.
-    isClosing = true
+    closedTime = Some(System.currentTimeMillis())
```

Comment:
Is is possible to guard `closedTime` with `executionsLock` rather than having to explain how this is correct?
(may be this field is used in many places).